tests/framework/BaseYiiTest.php

```diff
@@ -84,4 +86,19 @@ public function testCreateObjectCallable()
             return true;
         }));
     }
+
+    /**
+     * @covers yii\BaseYii::setLogger()
+     */
+    public function testSetLogger()
+    {
+        $logger = new Logger();
+        BaseYii::setLogger($logger);
+
+        $reflection = new \ReflectionClass('yii\BaseYii');
```

Comment:
The test will pass, but it will depend on the getLogger method. If the method getLogger change the logic, the test with name "setLogger" will fail.
Is it okay?